pkg/controller/controller.go

```diff
@@ -291,6 +293,11 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 	for k, v := range rep.Volume.Attributes {
 		volumeAttributes[k] = v
 	}
+	respCap := rep.GetVolume().GetCapacityBytes()
+	if respCap < volSizeBytes {
+		return nil, fmt.Errorf("created volume capacity %v less than requested capacity %v", respCap, volSizeBytes)
```

Comment:
This leaves a zombied volume. All returns after line 289 must delete the volume on error to clean up.

Reply:
Added logic to delete volume as well as unit test to check this behavior. We should be able to use the method I used to unit test this functionality for more unit tests on the Provisioner in the future (there are currently none)